__tests__/component_specs/calendar_page/cancellation_form_spec.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { shallow } from "enzyme";
+import sinon from 'sinon';
+
+import { ReservationForm } from "../../../app/javascript/components/calendar_page/reservation_form";
+import * as reservationActions from '../../../app/javascript/store/actions/reservationsActions';
+import * as planesActions from '../../../app/javascript/store/actions/planesActions';
+
+let form;
+describe('Reservation form', () => {
+    beforeAll(() => {
+        form = shallow(<ReservationForm planes={[{ id: 1, name: "something" }]}/>);
+    });
+
+    it('has a panel header with right props', () => {
+        expect(form.find('Panel').first().props().header).toEqual(<h3>Tee varaus</h3>);
+    });
+
+    it('has correct amount of form groups', () => {
+        expect(form.find('FormGroup').length).toEqual(7);
+    });
+
+    it('has select for planes', () => {
+        form.setProps({selectedPlane: { id: 2, name: "something else!"}});
+        form.update();
+        expect(form.find('FormControl').first().props().componentClass).toEqual("select");
+        expect(form.find('FormControl').first().props().value).toEqual(2);
+    });
+
+    it('has select with correct options for selecting plane', () => {
+        expect(form.find('FormGroup#selectPlane > FormControl > option').length).toEqual(2);
+        expect(form.find('FormGroup#selectPlane > FormControl > option').first().props().value).toEqual('null');
+        expect(form.find('FormGroup#selectPlane > FormControl > option').at(1).props().value).toEqual(1);
+    });
+
+    it('has Button with type submit', () => {
+        expect(form.find('Button').length).toEqual(1);
+        expect(form.find('Button').first().props().type).toEqual('submit');
+    });
+
+    it('has DatePickers', () => {
+        expect(form.find('#startDate').length).toEqual(1);
+        expect(form.find('#endDate').length).toEqual(1);
+    });
+
+    it('has two TimePickers', () => {
+        expect(form.find('#startTime').length).toEqual(1);
+        expect(form.find('#endTime').length).toEqual(1);
+    });
+
+    describe('dispatches correct actions', () => {
+        let actionStub, dispatchSpy;
+
+        beforeAll(() => {
+            dispatchSpy = sinon.spy();
+            form.setProps({dispatch: dispatchSpy});
```

Comment:
Kun muutat komponenttia tälläsisstä shallow-renderöidyissä testeissä niin sen jälkeen pitäis kutsua wrapperille myös update(), että komponentti käyttää noita uusia arvoja. Eli form.update()